### PR TITLE
Http misra/fix violations

### PR DESCRIPTION
**Pull request by aggarw13**
*Description of changes:*
Address MISRA violations
Replace use of `snprintf` with `memcpy` for MISRA compliance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/http/src/http_client.c [hunk 1]
```diff
@@ -108,19 +107,80 @@ static HTTPStatus_t _getFinalResponseStatus( HTTPParsingState_t parsingState,
...
+ *
+ * @return Returns the number of bytes written to @p pBuffer.
+ */
+static uint8_t _convertInt32ToAscii( int32_t value,
+                                     char * pBuffer,
+                                     size_t bufferlength );
+
+/*-----------------------------------------------------------*/
+
+static uint8_t _convertInt32ToAscii( int32_t value,
```
Comment (yourslab) [inline libraries/standard/http/src/http_client.c:124]: Couldn't we also use itoa and define a separate method for returning the length of an int32_t?

Comment (aggarw13) [reply libraries/standard/http/src/http_client.c:124]: `itoa` is not part of ANSI C as mentioned under the Portability section of its reference doc: http://www.cplusplus.com/reference/cstdlib/itoa/

We only want to add dependencies on ANSI C defined functions

File: libraries/standard/http/src/http_client.c [hunk 2]
```diff
@@ -1,6 +1,4 @@
-#include <assert.h>
```
Comment (sarenameas) [inline libraries/standard/http/src/http_client.c:1]: Why is this deleted? Where will this header exist?

File: libraries/standard/http/src/private/http_client_internal.h [hunk 3]
```diff
@@ -1,6 +1,16 @@
 #ifndef HTTP_CLIENT_INTERNAL_H_
 #define HTTP_CLIENT_INTERNAL_H_
 
+#include "config.h"
+
+#if defined( HTTP_ENABLE_ASSERT ) && ( HTTP_ENABLE_ASSERT == 1 )
```
Comment (sarenameas) [inline libraries/standard/http/src/private/http_client_internal.h:6]: I think we shouldn't have this and instead if customers are going to compile out asserts they will use the ANSI C definition NDEBUG in their compiler level macros.

File: libraries/standard/http/src/http_client.c (hunk 2 above)
Comment (aggarw13) [reply libraries/standard/http/src/http_client.c:1]: `http_client_internal.h`

File: libraries/standard/http/src/private/http_client_internal.h (hunk 3 above)
Comment (aggarw13) [reply libraries/standard/http/src/private/http_client_internal.h]: We discussed offline, and decided to defer the decision on supporting configurable asserts per library in consultation with the larger team.
